3_javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,122 @@
+  
+  if (command.includes("what is my name")) {
+    if (name !== "") {
+      return `Your name is ${name}`;
+    } else {
+      return `You didn't tell me your name yet.`;
+    }
+  }
+  
+  // todo------------------------------------------------------------
+  if(command.includes("todo")){
+    if(command.includes('add')){
+        let addIndex = splittedCommand.indexOf('add');
+        let toIndex = splittedCommand.indexOf('to');
+        todoItem = splittedCommand.slice(addIndex + 1 , toIndex).join(' ')
+        todos.push(todoItem);
+        return `${todoItem} added to your todos `
+    }
+    if(command.includes('remove')){
+        let removeIndex = splittedCommand.indexOf('remove');
+        let fromIndex = splittedCommand.indexOf('from')
+        let itemToRemove = splittedCommand.slice(removeIndex + 1 , fromIndex).join(' ')
+        for( const todo of todos){
+            if(todo === itemToRemove){
+                todos.splice(todos.indexOf(todo), 1);
+                return `${todo} removed from your todo list.`
+            }else{
+                return `you didn't add ${itemToRemove} to your todo list.`
+            }
+        }
+    }
+    if(command === "waht is on my todo" || command === "show my todo list" ){
+        let todoList = ""
+        for(const todo of todos){
+            todoList += `${todo}\n ` ;
+        }
+        return todoList;
+    }
+
+  }
+
+//   date ------------------------------------------
+    if(command.includes("what day is today")){
+        const today = new Date();
+        const options = {day: 'numeric' , month: 'long' , year: 'numeric'};
+        const formatedDate = today.toLocaleDateString('en-US', options)
+        return `today is ${formatedDate}`
+    }
+// math--------------------------------------------------------------
+
+    if(command.includes("what is")){
+        let mathResult;
+        let num1 = parseInt(splittedCommand[2]);
+        let num2 = parseInt(splittedCommand[4]);
+        let oprator = splittedCommand[3]
+        switch (oprator){
+            case '+':
+                mathResult = num1 + num2;
+                break;
+            case '-' :
+                mathResult = num1 - num2;
+                break;
+            case '*' :
+                mathResult = num1 * num2;
+                break;
+            case '/':
+                mathResult = num1 / num2;
+                break;
+            default:
+                return "Invalid operator. Supported operators are +, -, *, /";
+        }
+        return `the result is ${mathResult}`;
+           
+    }
+    // timer--------------------------------------------------------
+    if (command.includes("set timer for")) {
+        const forIndex = splittedCommand.indexOf('for');
+        const time = parseInt(splittedCommand[forIndex + 1]);
+        setTimeout(() => {
+            console.log('Timer done');
+        }, time * 1000 * 60 ); 
+    
+        return(`Timer set for ${time} minutes`) ;  
+    }
+};
```

Comment:
To stay at safe side, you can return a string here, that will show at any case that is not defined by you inside the function, fx:
now if you use any wrong command, it will return `undefined`
so I am suggesting something like this:

```suggestion
 return "I don't understand you";
```